### PR TITLE
Added update email mutation to intranet schema

### PR DESCRIPTION
**Pull request by BartoszJarocki**
Related issue: https://github.com/ParabolInc/parabol/issues/5005 

One note, `User` is also updated in PG, I'm not sure about `TeamMember` as I don't see what's the relation between `User` and `Team` in PG so I haven't done anything related to that.

### CODE REVIEW

File: packages/server/graphql/intranetSchema/mutations/updateEmail.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  },
+  resolve: async (_source, {oldEmail, newEmail}, {authToken}: InternalContext) => {
+    const r = await getRethink()
+
+    // AUTH
+    requireSU(authToken)
+
+    // VALIDATION
+    if (!newEmail) {
+      throw new Error('New email can not be empty')
```
Comment (mattkrick) [inline packages/server/graphql/intranetSchema/mutations/updateEmail.ts:28]: `type: GraphQLNonNull(GraphQLString),` already validates that the email is nonnull so no need to repeat here

File: packages/server/graphql/intranetSchema/mutations/updateEmail.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    }
+
+    if (oldEmail === newEmail) {
+      throw new Error('New email is the same as the old one')
+    }
+
+    const email = newEmail.toLowerCase().trim()
```
Comment (mattkrick) [inline packages/server/graphql/intranetSchema/mutations/updateEmail.ts:35]: let's normalize both emails before the validation begins

File: packages/server/graphql/intranetSchema/mutations/updateEmail.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+      .getAll(oldEmail, {index: 'email'})
+      .nth(0)
+      .default(null)
+      .run()
+    if (!user) {
+      throw new Error(`User with ${oldEmail} not found`)
+    }
+
+    // RESOLUTION
+    const userId = user.id
```
Comment (mattkrick) [inline packages/server/graphql/intranetSchema/mutations/updateEmail.ts:51]: this is gonna be the absolutely most nittiest of requests & if you don't like it i'm totally OK with that :see_no_evil: 
when possible, could you destructure like `const {id: userId} = user`? 
Even though it's a little bit longer, it means fewer line diffs so in the future if we need to get more props out of that object e.g. `const {id: userId, createdAt, updatedAt} = user` is just gonna look like a partial line diff

File: packages/server/graphql/intranetSchema/mutations/updateEmail.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+import {requireSU} from '../../../utils/authorization'
+import {InternalContext} from '../../graphql'
+import updateUser from '../../../postgres/queries/updateUser'
+
+const updateEmail = {
+  type: GraphQLNonNull(GraphQLBoolean),
+  description: `Updates the user email`,
+  args: {
+    oldEmail: {
+      type: GraphQLNonNull(GraphQLString),
```
Comment (mattkrick) [inline packages/server/graphql/intranetSchema/mutations/updateEmail.ts:12]: if you change this to a GraphQLID then graphql will validate that it doesn't contain any spaces

**Review by mattkrick — COMMENTED** [review]
looking almost perfect! just a couple nits

File: packages/server/graphql/intranetSchema/mutations/updateEmail.ts (hunk 3 above)
Comment (BartoszJarocki) [reply packages/server/graphql/intranetSchema/mutations/updateEmail.ts:51]: I actually think that's a very valid request, fixed 😅

File: packages/server/graphql/intranetSchema/mutations/updateEmail.ts (hunk 4 above)
Comment (BartoszJarocki) [reply packages/server/graphql/intranetSchema/mutations/updateEmail.ts:12]: @mattkrick I see we have https://github.com/ParabolInc/parabol/blob/8dbff9b4c6d19467c96b323be63f61e2926dc0a5/packages/server/graphql/types/GraphQLEmailType.ts, maybe that's the best of both worlds? I see it's not trimming the value but I can add that and then it'd be consistent across the codebase. Also, all the checks would be explicit which I think is a bit more readable. What do you think?

File: packages/server/graphql/intranetSchema/README.md [hunk 5]
```diff
@@ -0,0 +1,11 @@
+## Intranet/Admin interface 
+
+This part of Parabol's API is available only for admins. To become an admin user has to have `su` role set.
+
+`su` role can be set via RethinkDB’s data explorer, on local environment it’s available at: http://localhost:8080/#dataexplorer
+```
+r.db('actionDevelopment').table('User').filter({email: 'set-your-email-here'}).update({rol:'su'})
```
Comment (mattkrick) [inline packages/server/graphql/intranetSchema/README.md:7]: could you add the SQL here, too? want to keep the User data across both DBs aligned

File: packages/server/graphql/intranetSchema/mutations/updateEmail.ts (hunk 4 above)
Comment (mattkrick) [reply packages/server/graphql/intranetSchema/mutations/updateEmail.ts:12]: oh yeaaah forgot about that type! no strong preference either way, whatever you think looks best

File: packages/server/graphql/intranetSchema/mutations/updateEmail.ts (hunk 1 above)
Comment (BartoszJarocki) [reply packages/server/graphql/intranetSchema/mutations/updateEmail.ts]: Fixed 👍

File: packages/server/graphql/intranetSchema/mutations/updateEmail.ts (hunk 2 above)
Comment (BartoszJarocki) [reply packages/server/graphql/intranetSchema/mutations/updateEmail.ts]: Fixed by using `GraphQLEmailType` which doesn't allow spaces and makes the value lowercase.

File: packages/server/graphql/intranetSchema/mutations/updateEmail.ts (hunk 4 above)
Comment (BartoszJarocki) [reply packages/server/graphql/intranetSchema/mutations/updateEmail.ts]: Changed to `GraphQLEmailType`. I think it should be consistent across the codebase.

File: packages/server/graphql/intranetSchema/README.md (hunk 5 above)
Comment (BartoszJarocki) [reply packages/server/graphql/intranetSchema/README.md:8]: 🙈  I forgot about the PG, and sure, added!

**Comment by BartoszJarocki** [comment]
@mattkrick I'd appreciate if you could take a look again, thank you!

**Review by tianrunhe — APPROVED** [review]
LGTM!

**Comment by mattkrick** [comment]
nice one! this will help the service folks a bunch!

**Pull request merged**